apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KappControllerConfigSpec defines the desired state of KappControllerConfig
+type KappControllerConfigSpec struct {
+	// The namespace in which kapp-controller is deployed
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=kube-system
+	Namespace string `json:"namespace,omitempty"`
+
+	KappController KappController `json:"kappController,omitempty"`
+}
+
+type KappController struct {
+	// Whether to create namespace specified for kapp-controller
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	CreateNamespace bool `json:"createNamespace,omitempty"`
+
+	// The namespace value used for global packaging resources. Any Package and PackageMetadata CRs within that namespace will be included in all other namespaces on the cluster, without duplicating them
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	GlobalNamespace string `json:"globalNamespace,omitempty"`
+
+	Deployment KappDeployment `json:"deployment,omitempty"`
+
+	Config KappConfig `json:"config,omitempty"`
+}
+
+type KappDeployment struct {
+	// Whether to enable host networking for kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+
+	// The priority value that various system components use to find the priority of the kapp-controller pod
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=system-cluster-critical
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// Concurrency of kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=4
+	Concurrency int `json:"concurrency,omitempty"`
+
+	// kapp-controller deployment tolerations
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:={{key: CriticalAddonsOnly, operator: Exists}, {effect: NoSchedule, key: node-role.kubernetes.io/master}, {effect: NoSchedule, key: node.kubernetes.io/not-ready}, {effect: NoSchedule, key: node.cloudprovider.kubernetes.io/uninitialized, value: "true"}}
+	Tolerations []map[string]string `json:"tolerations,omitempty"`
+
+	// Bind port for kapp-controller API
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=10350
+	APIPort int `json:"apiPort,omitempty"`
+
+	// Address for metrics server
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=:8080
```

Comment:
default is `0` right?

Reply:
https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml#L17
https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml#L34
It was defaulted to `:8080` in TCE.

Reply:
Right but TCE doesn't deploy kapp-controller using that config. We disabled it in TKG because it runs on hostNetwork and don't want to expose metrics without any auth.

Reply:
Given this, do we need to change the default of `hostNetwork` to true?